hack/generator/pkg/codegen/testdata/AllOf/AllOf_turns_map_type_back_into_additionalProperties.golden

```diff
@@ -0,0 +1,12 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// Code generated by k8s-infra-gen. DO NOT EDIT.
+package v20200101
+
+//Generated from: https://test.test/schemas/2020-01-01/test.json
+type Test struct {
+
+	// +kubebuilder:validation:Required
+	Size                 int             `json:"size"`
+	additionalProperties map[string]bool `json:"additionalProperties"`
```

Comment:
Should this be exported? As it stands client code couldn't construct one with a passed-in map.

Reply:
The support for it is still incomplete. At the moment we don’t handle this properly at all and this needs to be addressed in the serialization layer, so I’ve created an issue for this: https://github.com/Azure/k8s-infra/issues/187

Reply:
Ah, ok - so we're only expecting something like this to be coming back from ARM as part of status, rather than expecting users to construct them?

Reply:
we will allow it in future but for now this doesn‘t work at all, so…